x-pack/platform/plugins/shared/fleet/server/services/epm/packages/get.ts

```diff
@@ -538,14 +567,52 @@ export async function getPackageInfo({
     licensePath: Registry.getLicensePath(paths || []),
     keepPoliciesUpToDate: savedObject?.attributes.keep_policies_up_to_date ?? false,
   };
-  const updated = { ...packageInfo, ...additions };
+
+  const { filteredDataStreams, filteredPolicyTemplates } =
+    getFilteredDataStreamsAndPolicyTemplates(packageInfo);
+
+  const updated = {
+    ...packageInfo,
+    ...additions,
+    data_streams: filteredDataStreams,
+    policy_templates: filteredPolicyTemplates,
+  };
 
   const installable = createInstallableFrom(updated, savedObject);
   setPackageInfoCache(pkgName, pkgVersion, installable);
 
   return installable;
 }
 
+function getFilteredDataStreamsAndPolicyTemplates(packageInfo: ArchivePackage | RegistryPackage) {
+  const excludeDataStreamTypes =
+    appContextService.getConfig()?.internal?.excludeDataStreamTypes ?? [];
+  let filteredDataStreams = packageInfo.data_streams;
+  let filteredPolicyTemplates = packageInfo.policy_templates;
+
+  if (excludeDataStreamTypes.length > 0) {
+    filteredDataStreams = packageInfo.data_streams?.filter(
+      (dataStream) => !excludeDataStreamTypes.includes(dataStream.type)
+    );
+    // filter out matching types e.g. nginx/metrics
+    filteredPolicyTemplates = packageInfo.policy_templates?.reduce(
+      (acc: any[], policyTemplate: any) => {
```

Comment:
Can the type for `policyTemplate`  be specified? I think it should be this one: https://github.com/elastic/kibana/blob/8d0883533d70bdfee72b60e8b99e5dd181aa2f1e/x-pack/platform/plugins/shared/fleet/common/types/models/package_spec.ts#L28

Reply:
updated